### ISSUE TITLE
ActiveRecord::Type::Boolean doesn't return false on string "False"

### ISSUE THREAD

**Issue by colto** [rails/rails#39069]
### Steps to reproduce
Pass string `"False"` to `ActiveRecord::Type::Boolean.new.cast`

### Expected behavior
```ruby
DEVELOPMENT [1]> ActiveRecord::Type::Boolean.new.cast("False")
=> false
```

### Actual behavior
```ruby
DEVELOPMENT [1]> ActiveRecord::Type::Boolean.new.cast("False")
=> true
```

### System configuration
**Rails version**: `6.0.2.1`

**Ruby version**: `ruby 2.6.5p114 (2019-10-01 revision 67812) [x86_64-darwin19]`

**Comment by jonathanhefner** [comment]
Would you mind sharing some context, i.e. where is the string `"False"` coming from?

Currently, both `"FALSE"` and `"false"` are cast to `false`, so casting `"False"` seems reasonable.  But I don't think we want to support additional variants just for the sake of doing so.  For example, we will probably not add support for `"fALSE"`.

**Comment by colto** [comment]
@jonathanhefner Totally. In our app, we have a few flags that we are expecting for our clients to send and we have implemented a `to_boolean` initializer that extends String using `ActiveModel::Type::Boolean.new.cast(self)` - it's easy enough for us to say `ActiveModel::Type::Boolean.new.cast(self&.downcase)` but I figured why not try to tackle it at the root.

It looks like there is quite an extensive array of values, and I understand just adding a value or two at a time, but my thought was why not coerce the value being passed in to `cast_value` into being a lowercase string. Then the `FALSE_VALUES` array can be simplified to a few values that (regardless of case) will return false.

[Here is my PR](https://github.com/rails/rails/pull/39071)
